### ISSUE TITLE
Add support for italics

### ISSUE THREAD

**Issue by E5ten** [tcr/rager#8]
When italics are passed to rager it gives the error `Unknown CSI 'm' param 3` and does not successfully display italics.

**Comment by tcr** [comment]
You're right. I think this issue is actually upstream, and it looks like the `ransid` crate (part of Redox OS) skips over the Italic CSI "3" and thus doesn't have an API option to decode it: https://github.com/redox-os/ransid/blob/18a50799d01e28e0d74db182c1ed3a77e9f0c1a9/src/lib.rs#L562 If that were fixed, it could be supported in rager. 

Thanks for filing an issue!

**Comment by E5ten** [comment]
Until ransid supports italics, would it be possible for rager to intercept calls for italics and make them do nothing instead of causing errors?

**Comment by E5ten** [comment]
Now I don't really know rust, so I'm sure I probably did something that isn't very good in my code, but I did fork ransid and just basically replicated what it does with the word underlined but with italic, and had the param 3 use that, and then I edited rager a bit and did the same thing, so I'll make a PR with my rager edits and a change to the Cargo.toml to use my fork through git, I understand if you want to only use upstream ransid though.
EDIT: Nevermind for now, it's not working properly, I'll keep trying to work on it

**Comment by E5ten** [comment]
Ok, it might actually have been working? so there's no error, CSI param 3 is recognized by ransid in my fork and should do italics, rager passes italics through and has a bool for it, but in rager the text isn't actually italicized despite italics working in my terminal, however, the same occurs with underlined text, so my changes might be working correctly but for some other reason rager isn't able to display italicized or underlined text, instead displaying it as normal text.

**Comment by tcr** [comment]
@E5ten Looks likes rancid needs a way to disable spurious logging, it also needs support for italics, but underlining and italics were also not rendered being in rager.

Can I point rager to rely on your fork for the moment? I think we can rely on a forked version temporarily and upstream the fixes to ransid.  If you wanted to add my changes to [switch to the log crate for errors](https://github.com/tcr/ransid/commit/998c158cc816576719ac29652ca67585edc8bc44) I think this adds the common cases and future-proofs others.

As for rager, it looks like I added bold support and just completely missed underlining support. It's easy enough to add here underneath the check for bold: https://github.com/tcr/rager/blob/master/src/main.rs#L156 Is this something you'd be interested in writing a patch for + adding support for italics? 

[From stackoverflow](https://askubuntu.com/a/528938) I found an easy test for different types of shell escapes, I can imagine adding this to a unit test case at some point:

```
echo -e "\e[1mbold\e[0m" # shows up in bold through rager
echo -e "\e[3mitalic\e[0m" # normal
echo -e "\e[4munderline\e[0m" # normal (bug)
echo -e "\e[9mstrikethrough\e[0m" # normal
echo -e "\e[31mHello World\e[0m"  # red text
echo -e "\x1B[31mHello World\e[0m" # red text
```
